lib/galaxy/model/migrations/__init__.py

```diff
@@ -253,10 +253,10 @@ def _load_sqlalchemymigrate_version(self, conn: Connection) -> CursorResult:
 
 def metadata_contains_only_kombu_tables(metadata: MetaData) -> bool:
     """
-    Return True if metadata contains exactly 2 tables: kombu_message and kombu_queue.
+    Return True if metadata contains only kombu-related tables.
     (ref: https://github.com/galaxyproject/galaxy/issues/13689)
     """
-    return len(metadata.tables) == 2 and "kombu_message" in metadata.tables and "kombu_queue" in metadata.tables
+    return all(table.startswith("kombu_") or table.endswith("_sequence") for table in metadata.tables.keys())
```

Comment:
I think this is too broad. We know exactly the tables: 2 kombu tables + an internal sqlite table - why not just check for these 3?

Reply:
Future proofing in case celery adds more kombu tables. What's the downside?

Reply:
Premature optimization? yagni? Not impossible that galaxy might have a table that fits that pattern? If you feel it's a better approach, let's do it, but I would still suggest we explicitly check for `sqlite_sequence`.

Reply:
Backing up a step, what are we trying to do in the larger context here?  Just make sure Galaxy hasn't already initialized in this database, right?   Would it make more sense to just check to see if `galaxy_user` or other table(s) exist, and if not, move forward?

Reply:
#13689: celery starts (or may start) before galaxy, so the db is not empty during a new install, so the algorithm that reasons about the state of the database gets confused (see whiteboard, top left corner #13108). Checking for an early galaxy table has the same potential issue as checking for a pattern in the table name.

Reply:
Thanks for the context.  I guess what I was getting at was -- we don't care that the database has exactly two (or three) tables.  We don't have to consider what else might be in there at all -- we just want to know if Galaxy (for core or kombu) has done its initialization, and we can (I think) trust admins a bit more.  We can refactor these checks to see:

1) do we need to initialize galaxy db?  (check for early galaxy table)
   If yes, then initialize galaxy.
2) do we need to initialize kombu tables?  (check for kombu tables from that migration -- not all, just something indicative).
    If yes, then initialize kombu.

Reply:
Just found that there are a bunch of `sqlite_` special tables ( https://www.sqlite.org/fileformat2.html#storage_of_the_sql_database_schema ), so I'm thinking to change this to:

```suggestion
    return all(table.startswith("kombu_") or table.startswith("sqlite_") for table in metadata.tables.keys())
```

Reply:
We don't initialize kombu - celery takes care of that. I think checking for `kombu_*` and `sqlite_*` is the most straightforward  and simple approach.

Reply:
This tweak is fine with me but I think my point that we can probably refame the questions the stack is asking at this point is reasonable.  We shouldn't have to care about `contains_only_kombu_tables` -- we just want to know if the things Galaxy requires need initialization, whichever components are responsible for those things.

Reply:
There's more to it: there may be one or two databases: galaxy and/or tool shed install, one may have been initialized, the other one hasn't; it may have been one combined database which is now being switched to 2 separate databases, or vise versa. There are edge cases that emerge based on configuration and on how the upgrade process is carried out. I think, checking at this point for specific tables (or groups of tables) is simple and straightforward. I may be wrong, of course.